dns-integrator-operator/src/charm.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""DNS Integrator Charm."""
+
+import logging
+import typing
+import hashlib
+import uuid
+
+import ops
+from charms.bind.v0 import dns_record
+
+logger = logging.getLogger(__name__)
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+
+
+class DnsIntegratorOperatorCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args: typing.Any):
+        """Construct.
+
+        Args:
+            args: Arguments passed to the CharmBase parent constructor.
+        """
+        super().__init__(*args)
+        self.dns_record = dns_record.DNSRecordRequires(self)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
+        """Handle changes in configuration."""
+        self.unit.status = ops.MaintenanceStatus("Configuring charm")
+        logger.debug("UPDATE INTEGRATOR CONFIG")
+        self._update_relations()
+        self.unit.status = ops.ActiveStatus()
+
+    def _update_relations(self) -> None:
+        """Update all DNS data for the existing relations."""
+        if not self.model.unit.is_leader():
+            return
+        try:
+            for relation in self.model.relations[self.dns_record.relation_name]:
+                self.dns_record.update_relation_data(
+                    relation, self._get_dns_record_requirer_data()
+                )
+        except ops.model.ModelError as e:
+            logger.error("ERROR while updating relation data: %s", e)
+
+    def _get_dns_record_requirer_data(self) -> dns_record.DNSRecordRequirerData:
+        """Get DNS record requirer data."""
+        entries = []
+        for request in str(self.config["requests"]).split("\n"):
+            data = request.split()
+            if len(data) != 6:
```

Comment:
Should 6 be a constant?

Reply:
There are exactly 6 fields that I'm waiting for. Those are listed two lines further down

Reply:
I was thinking about the case of data changes in the future but no strong opinion about that.